app/board.js

```diff
@@ -1,53 +1,97 @@
 import { Enum } from './enum'
-import { Players } from './players'
+import { Players } from './enum'
 
 class Board {
 
-  constructor() {
-    this.cells = new Array(Enum.BOARD_ROWS_NUMBER)
-    this.cleanBoard()
-    this.currentPlayer = Players.NONE
-  }
+  constructor(cells, player, size) {
+    if (cells == null){
+      let _emptyMapObj = {}
+      for (let i=0; i<Enum.BOARD_COLUMNS_NUMBER; i++)
+      for (let j=0; j<Enum.BOARD_ROWS_NUMBER; j++) {
+        let key = this.formatKey(i,j)
+        _emptyMapObj[key] = Players.NONE
+      }
+      this.cells = new Immutable.Map(_emptyMapObj)
+    } else {
+      this.cells = cells
+    }
 
+    let _currentPlayer = player
+    if (player == null) {
+      _currentPlayer = Math.round(Math.random())
+    }
+    this.currentPlayer = () => { return _currentPlayer }
```

Comment:
`let currentPlayer = player || Math.round(Math.random())`
`this.currentPlayer = () => {  return currentPlayer }`